searchcore/src/vespa/searchcore/proton/feedoperation/removeoperation.h

```diff
@@ -7,22 +7,57 @@
 namespace proton {
 
 class RemoveOperation : public DocumentOperation {
+protected:
+    explicit RemoveOperation(Type type) : DocumentOperation(type) {}
+    RemoveOperation(Type type, document::BucketId bucketId, storage::spi::Timestamp timestamp)
+        : DocumentOperation(type, bucketId, timestamp)
+    {}
+public:
+    virtual bool hasDocType() const = 0;
+    virtual vespalib::stringref getDocType() const = 0;
+    virtual const document::GlobalId & getGlobalId() const = 0;
+};
+
+class RemoveOperationWithDocId : public RemoveOperation {
     document::DocumentId _docId;
 
 public:
-    RemoveOperation();
-    RemoveOperation(const document::BucketId &bucketId,
-                    const storage::spi::Timestamp &timestamp,
-                    const document::DocumentId &docId);
-    virtual ~RemoveOperation() {}
+    RemoveOperationWithDocId();
+    RemoveOperationWithDocId(document::BucketId bucketId,
+                             storage::spi::Timestamp timestamp,
+                             const document::DocumentId &docId);
+    ~RemoveOperationWithDocId() override;
     const document::DocumentId &getDocumentId() const { return _docId; }
-    virtual void serialize(vespalib::nbostream &os) const override;
-    virtual void deserialize(vespalib::nbostream &is,
-                             const document::DocumentTypeRepo &repo) override;
-    virtual vespalib::string toString() const override;
+    const document::GlobalId & getGlobalId() const override { return _docId.getGlobalId(); }
+    void serialize(vespalib::nbostream &os) const override;
+    void deserialize(vespalib::nbostream &is, const document::DocumentTypeRepo &repo) override;
+    vespalib::string toString() const override;
+
+    bool hasDocType() const override { return _docId.hasDocType(); }
+    vespalib::stringref getDocType() const override { return _docId.getDocType(); }
+};
+
+class RemoveOperationWithGid : public RemoveOperation {
+    document::GlobalId _gid;
+    vespalib::string   _docType;
+    uint32_t           _lid;
```

Comment:
Old lid already exists in grandparent class, c.f. getPrevLid() method.

Reply:
OK, revmoved